crates/ruff_linter/src/rules/flake8_comprehensions/rules/unnecessary_comprehension.rs

```diff
@@ -140,10 +146,80 @@ pub(crate) fn unnecessary_list_set_comprehension(
     if !generator.ifs.is_empty() || generator.is_async {
         return;
     }
-    if ComparableExpr::from(elt) != ComparableExpr::from(&generator.target) {
-        return;
+    if is_dict_items(checker, &generator.iter) {
+        match (&generator.target, elt) {
+            // [(k, v) for k, v in dictionary.items()]j
+            (
+                Expr::Tuple(ast::ExprTuple {
+                    elts: target_elts, ..
+                }),
+                Expr::Tuple(ast::ExprTuple { elts, .. }),
```

Comment:
Do we need to check for list as well?

Reply:
Yes, `target` can be a list. Thanks for the catch :)